### PR TITLE
Adds azure as a push location to the docker exporter

### PR DESCRIPTION
**Pull request by eeyun**
Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/pkg-export-docker/src/lib.rs [hunk 1]
```diff
@@ -148,6 +148,16 @@ impl Credentials {
 
                 Ok(Credentials { token: token })
             }
+            RegistryType::Azure => {
```
Comment (elliott-davis) [inline components/pkg-export-docker/src/lib.rs]: Can this say 
```
RegistryType::Azure | RegistryType::Docker => {
                Ok(Credentials {
                    token: base64::encode(&format!(
                        "{}:{}",
                        username.to_string(),
                        password.to_string()
                    )),
                })
}
```

**Comment by eeyun** [comment]
Notes on this PR:
Azure Container Registry is a pretty lightweight registry that leans heavily on the same pattern as private vanilla docker registries with one important exception. This registry is tied directly into Azure's AD and IAM account services. 

What this means from a user perspective is that in order to use the azure exporter endpoint, the user must have configured an AD Service Principal for the registry. Without a service prinicipal account the user would need to log into azure with an azure cli and IAM role. In the context of the azure docs its suggested that any interactions with ACR driven by automation leverage said [service principal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal).

File: components/pkg-export-docker/src/lib.rs (hunk 1 above)
Comment (eeyun) [reply components/pkg-export-docker/src/lib.rs]: Oh yeah it most definitely can.

**Comment by raskchanky** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/356037609).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/356037609).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**